Sources/ComposableArchitecture/Effect.swift

```diff
@@ -411,14 +418,55 @@ public struct Send<Action> {
   /// - Parameters:
   ///   - action: An action.
   ///   - transaction: A transaction.
-  public func callAsFunction(_ action: Action, transaction: Transaction) {
-    guard !Task.isCancelled else { return }
-    withTransaction(transaction) {
+  @discardableResult
+  public func callAsFunction(_ action: Action, transaction: Transaction) -> SendTask {
+    guard !Task.isCancelled else { return .init(rawValue: nil) }
+    return withTransaction(transaction) {
       self(action)
     }
   }
 }
 
+/// The type returned from ``Send/callAsFunction(_:)`` that represents the lifecycle of the effect
+/// started from sending an action inside ``EffectPublisher/run(priority:operation:catch:file:fileID:line:)``.
+///
+/// You can use this value to tie the Effect's lifecycle _and_ cancellation to another Effect.
+///
+/// ```swift
+/// return .run { send in
+///   await send(.anotherAction).finish()
+///   print("Done!")
+/// }
+/// ```
+///
+/// > Note: Unlike Swift's `Task` type, ``SendTask`` automatically sets up a cancellation
+/// > handler between the current async context and the task.
+///
+/// See ``TestStoreTask`` for the analog returned from ``TestStore``, and ``ViewStoreTask``
+/// for the analog returned from ``ViewStore``.
+public struct SendTask: Hashable, Sendable {
```

Comment:
I wonder whether we should unify `ViewStoreTask` and `SendTask` into a single type, just didn't want to prematurely apply DRY here.